src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/ParamServer.java

```diff
@@ -180,52 +197,65 @@ private void setFinishedState(int workerID) {
 			_finishedStates[workerID] = true;
 		}
 
+		private void broadcastModel() {
+			IntStream.range(0, _finishedStates.length).forEach(i -> _modelMap.compute(i, (id, q) -> {
+				if (q == null) {
+					q = new ArrayBlockingQueue<>(1);
+				}
```

Comment:
No need to create a new queue here - please simply put the new model into the queue.